### DISCUSSION TITLE
测试doris on es功能遇到的问题

### DISCUSSION THREAD

**Discussion by Allen-zy** [apache/doris#6678] [A - General / Q&A]
由于公司业务场景需要，既要保留ES的分词检索及模糊查询，又要具备大表关联查询，在测试Doris On ES功能时遇到一下问题：
Q1:select a.*,b.prod_desc,b.f_catalog_name,b.fromtype_desc 
from es_db.ads_sys_delivery_audited_2_uat a JOIN es_db.ads_owner_delivery_1_uat b 
on a.delivery_id = b.delivery_id 
where a.delivery_id = 11859065
查询时间：13ms
Q2:select a.*,b.prod_desc,b.f_catalog_name,b.fromtype_desc 
from es_db.ads_sys_delivery_audited_2_uat a JOIN es_db.ads_owner_delivery_1_uat b 
on a.delivery_id = b.delivery_id 
where a.delivery_num = "DD200711108976" 
查询时间：56s 
Q2:select a.*,b.prod_desc,b.f_catalog_name,b.fromtype_desc 
from es_db.ads_sys_delivery_audited_2_uat a JOIN es_db.ads_owner_delivery_1_uat b 
on a.delivery_num = b.delivery_num 
where a.delivery_num = "DD200711108976" 
查询时间：12ms
查询结果为同一条数据，经排查发现Q2的谓词条件没有下推到es，当Q2换成Q3时,该谓词条件又完成下推，希望大佬们指教一下

**Comment by morningman** [comment]
cc @wuyunfeng @qidaye

**Comment by dataalive** [comment]
sorry for so long no-response, we now start to debug this problem

**Comment by wuyunfeng** [comment]
hello啊，你把三个query：Q1 Q2 Q3 执行的时候 在be里面搜索Generated ES queryDSL [xxxxxxxx]，分别给贴一下？

**Comment by wuyunfeng** [comment]
Q2:select a.,b.prod_desc,b.f_catalog_name,b.fromtype_desc
from es_db.ads_sys_delivery_audited_2_uat a JOIN es_db.ads_owner_delivery_1_uat b
on a.delivery_id = b.delivery_id
where a.delivery_num = "DD200711108976"
这个肯定全表扫描的。。。
